plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2016, Peter Sagerson <psagers@ignorare.net>
+# Copyright: (c) 2020, Sebastian Pfahl <eryx@gmx.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r"""
+---
+module: ldap_search
+short_description: Search for entries in a LDAP server
+description:
+  - Return the results of an LDAP search. Use Ansible's 'register' statement.
```

Comment:
```suggestion
  - Return the results of an LDAP search.
```
Just a suggestion (because it's about how to use Ansible)